core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1347,7 +1355,7 @@ public CommitId flush(boolean force, boolean waitIfOngoing) throws EngineExcepti
                         commitIndexWriter(indexWriter, translog, null);
                         logger.trace("finished commit for flush");
                         // we need to refresh in order to clear older version values
-                        refresh("version_table_flush");
+                        refresh("version_table_flush"); // TODO technically we could also only refresh the internal searcher
```

Comment:
Doesn't the comment about releasing segments in tryRenewSyncCommit holds here too?

Reply:
I think I did some copy past mess here. I will fix